### ISSUE TITLE
TOML configuration language support

### ISSUE THREAD

**Issue by grinapo** [geany/geany#1268]
TOML is a fairly strictly defined configuration file syntax, used by various projects, supported by plenty of languages and already supported by some editors. I'm not familar with geany plugin dev (and lacking the time anyway) but maybe someone's interested in creating a syntax highlighting ruleset, similar to the INI config format.

https://github.com/toml-lang/toml

**Comment by biscottaro** [comment]
Mega necro bump, but I stumbled upon some toml files this week in unrelated projects. Could geany add this?

**Comment by elextr** [comment]
If "somebody" either:

a. made a [custom filetype](https://www.geany.org/manual/current/index.html#custom-filetypes) which needs an existing lexer which is "good enough" but anybody who wants it can do it, or 

b. created a new lexer and had it accepted at https://github.com/ScintillaOrg/lexilla and then made a PR to add it as part of a built-in filetype (C and C++ and Geany internals skills needed).

**Comment by rdipardo** [comment]
> If "somebody" either:
>
> [ . . . ]
>
> b. created a new lexer

Somebody downstream already has: https://github.com/zufuliu/notepad2/blob/main/scintilla/lexers/LexTOML.cxx

**Comment by elextr** [comment]
Well, now the creator of that needs to agree to change the license from Notepad's to Lexilla's to upstream it.  Luckily so far its only one person.

**Comment by rdipardo** [comment]
> Well, now the creator of that needs to agree to change the license from Notepad's to Lexilla's to upstream it.

Nah, he can generate more sponsorship dollars by reserving the rights. I'd say he's earned it. Have you *seen* [how active][0] that project is? A regular beehive.

[0]:https://github.com/zufuliu/notepad2/graphs/commit-activity

**Comment by elextr** [comment]
Well, [__HE__](https://github.com/zufuliu/notepad2/graphs/contributors) is active, others not so much.

Actually its not clear what license its under, the header refers to `License.txt` and there is one in the Scintilla directory immediately above the lexers directory and a different one in the top level directory, although that again says lexilla and scintilla is under their own license.  I guess the same could be done if Neil accepted it in Lexilla.

But anyway, if its not upstreamed its unlikely to get in Geany sadly.

**Comment by techee** [comment]
I just asked here https://github.com/zufuliu/notepad4/issues/806 and it's some custom lexer for Notepad4 only so we'd really have to implement the lexer by ourselves.

**Comment by elextr** [comment]
> we'd really have to implement the lexer by ourselves.

What you mean is that "somebody" would have to implement and submit the lexer to lexilla so others can share it.  It is precisely because various projects have personal lexers that they won't/can't share with upstream that such a common configuration language as TOML isn't supported.

**Comment by techee** [comment]
> What you mean is that "somebody" would have to implement and submit the lexer to lexilla so others can share it.

"Somebody" did: https://github.com/ScintillaOrg/lexilla/pull/261

See #3934.

**Comment by elextr** [comment]
> "Somebody" did

Good on "somebody" ;-P
